### ISSUE TITLE
Use tenacity (or similar) instead of re-implementing a retry function

### ISSUE THREAD

**Issue by marco-c** [mozilla/bugbug#1234]
We are currently implementing https://github.com/mozilla/bugbug/blob/c84c38f964aef514171adac10d54f4a33f3477e9/bugbug/utils.py#L202-L211 and using it in a few places.
We should instead use a library (e.g. tenacity) and remove this custom retry function.

**Comment by Pranay144** [comment]
```
def clone_git_repo(self, repo_url, repo_dir, rev="master", max_attempt=5):
        if not os.path.exists(repo_dir):
            tenacity.retry(
              lambda: subprocess.run(
                ["git","clone","--quiet",repo_url,repo_dir],
                check =True,
                stop = tenacity.stop_after_attempt(max_attempt)
              )
            )
```
@marco-c This is example code from 'commit_classifier.py line 191' with 'retry' function removed and implemented 'tenacity'. 
I want to ask if this code is right? so i can replace retry function from other places.

**Comment by marco-c** [comment]
@Pranay144 it's easier for me to review your code if you simply submit a PR. Don't worry if it's not perfect right from the start, you can make updates to the PR!
From a quick glance, it seems more or less correct (but we don't want to add the max_attempt parameter to clone_git_repo, and "stop" must be an argument of tenacity.retry not subprocess.run).

**Comment by amit6663** [comment]
can i work on this issue? @marco-c

**Comment by marco-c** [comment]
Anyone can work on any issue, read the guidance for first contributors, unless there is already a PR open to fix an issue.
@Pranay144 hasn't opened a PR yet, but it's almost as if he did, so he would have priority.

**Comment by marco-c** [comment]
Reopening as the PR is being reverted (as it breaks things).

**Comment by NitishNaineni** [comment]
@marco-c  i would like to work on this issue. Seems this issue was already fixed but is being reverted, what seems to be the issue here?

**Comment by marco-c** [comment]
@NitishNaineni usage of tenacity was not implemented correctly.

**Comment by aditya-hari** [comment]
@marco-c Would how it is implemented in mozilla/code-convergence#366 work?

**Comment by marco-c** [comment]
Yes
